modal/_runtime/asgi.py

```diff
@@ -283,6 +283,9 @@ def wait_for_web_server(host: str, port: int, *, timeout: float) -> None:
 async def _proxy_http_request(session: aiohttp.ClientSession, scope, receive, send) -> None:
     proxy_response: aiohttp.ClientResponse
 
+    if "client" in scope and "headers" in scope:
+        scope["headers"].append((b"X-Forwarded-For", scope["client"][0].encode()))
```

Comment:
why does `scope["client"][0]` have this header? maybe add a comment referring to the documentation or something, since this is highly nonobvious

Reply:
It's the asgi spec https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope

It doesn't have the header, it just has the source IP which this PR places in the header

Reply:
@ekzhang do you remember why we are not sending this as `X-Forwarded-For` from Caddy? 

Couldn't it make sense to have caddy set the `X-Forwarded-For` headers using `http.request.remote.host` too? (We shouldn't pass along a user-supplied such value, but we could make caddy set the header using its "trusted" info from the nlb?).
Not sure what the usage % of the scope["client"] value vs the standard header is among asgi apps in general, but it wouldn't hurt to fill both, right?

Reply:
@kramstrom regardless of caddy changes, it could make sense to check if the X-Forwarded-For header is already present in headers as well and just updating the existing header value in that case (if the scope value is already present in the X-Forwarded-For value, no change would be needed).